src/app/_hooks/useSearch.ts

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react'
+
+import { NextServerSearchParams } from '@/types/searchParams'
+import { SearchableByTitle } from '@/types/searchTypes'
+
+import { SEARCH_KEY } from '@/constants/searchParams'
+
+function validateSearchQuery(searchQuery: NextServerSearchParams['search']) {
+  if (!searchQuery) {
+    return ''
+  }
+
+  const rawQuery = Array.isArray(searchQuery) ? searchQuery[0] : searchQuery
+  return rawQuery.toLowerCase()
+}
+
+type UseSearchProps<Item extends SearchableByTitle> = {
+  searchParams: NextServerSearchParams
+  data: Item[]
+}
```

Comment:
Shall we move it up? I know it's closer like this - Just trying to be consistent for now.

Reply:
Also curious - Is there a preference for `Item` over `T`?

Reply:
`Item` feels slightly less cryptic than `T`?

Reply:
It is - and I think that's why it makes me stop and think - "What `Item` - and `Item` of what?" ... which isn't necessarily bad. Can we agree on `Entry`?

Reply:
✅